src/NCronJob/Registry/JobState.cs

```diff
@@ -0,0 +1,28 @@
+namespace NCronJob;
+
+internal class JobState
+{
+    public JobStateType Type { get; private set; }
+    public DateTime Timestamp { get; private set; }
+    public string Message { get; private set; }
+
+    public JobState(JobStateType type, string message = "")
+    {
+        Type = type;
+        Timestamp = DateTimeOffset.Now.LocalDateTime;
+        Message = message;
+    }
+}
+
+internal enum JobStateType
+{
+    Scheduled,
+    Running,
+    Retrying,
+    Completing,
+    Completed,
+    Failed,
```

Comment:
For the sake of consistency to .net itself, how about: `Faulted` as name? It is (currently) internal only anyway.

Reply:
Yes, of course.